### PR TITLE
Fyle import attributes

### PR DESCRIPTION
**Pull request by ruuushhh**

### CODE REVIEW

File: apps/fyle/models.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Reimbursement:
+    """
+    Creating a dummy class to be able to user
+    fyle_integrations_platform_connector correctly
+    """
+    pass
```
Comment (NileshPant1999) [inline apps/fyle/models.py:6]: new line

File: apps/fyle/serializers.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            refresh = self.context['request'].data.get('refresh')
+
+            workspace = Workspace.objects.get(id=workspace_id)
+            fyle_credentials = FyleCredential.objects.get(workspace_id=workspace_id)
+            platform = PlatformConnector(fyle_credentials)
+
+            if refresh:
+                platform.import_fyle_dimensions()
+                workspace.source_synced_at = datetime.now()
+                workspace.save(update_fields=['source_synced_at'])
```
Comment (Shwetabhk) [inline apps/fyle/serializers.py:33]: Always better to add line changes between each section for better readability

File: apps/fyle/serializers.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                    time_interval = datetime.now(timezone.utc) - workspace.source_synced_at
+                if workspace.source_synced_at is None or time_interval.days > 0:
+                    platform.import_fyle_dimensions()
+                    workspace.source_synced_at = datetime.now()
+                    workspace.save(update_fields=['source_synced_at'])
+            return Response(status=status.HTTP_200_OK)
+        except FyleCredential.DoesNotExist:
+            raise serializers.ValidationError({'message': 'Fyle credentials not found in workspace'}, code='invalid_login')
+        except Exception:
+            raise serializers.ValidationError()
```
Comment (Shwetabhk) [inline apps/fyle/serializers.py:45]: raise Exception?

also add loggers please

Everywhere the code can break should have loggers

**Review by Shwetabhk — APPROVED** [review]

**Review by NileshPant1999 — APPROVED** [review]

File: apps/fyle/models.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class Reimbursement:
+    """
+    Creating a dummy class to be able to user
```
Comment (NileshPant1999) [inline apps/fyle/models.py:3]: able to use 🤣

File: apps/fyle/serializers.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+from rest_framework import serializers
+from rest_framework.response import Response
+from rest_framework.views import status
+from fyle_integrations_platform_connector import PlatformConnector
+from datetime import datetime, timezone
+from apps.workspaces.models import Workspace, FyleCredential
+
+logger = logging.getLogger(__name__)
+logger.level = logging.INFO
+
```
Comment (NileshPant1999) [inline apps/fyle/serializers.py:14]: can we maintain 2 line spaces accorss repo

Comment (Shwetabhk) [reply apps/fyle/serializers.py:14]: +1

File: apps/fyle/serializers.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+                    workspace.save(update_fields=['source_synced_at'])
+
+            return Response(status=status.HTTP_200_OK)
+
+        except FyleCredential.DoesNotExist:
+            raise serializers.ValidationError({'message': 'Fyle credentials not found in workspace'}, code='invalid_login')
+
+        except Exception as exception:
+            logger.error('Something unexpected happened workspace_id: %s %s', workspace_id, exception)
+            raise serializers.ValidationError()
```
Comment (Shwetabhk) [inline apps/fyle/serializers.py:53]: we shouldn't raise validation error for broad exceptions, just raise exception only

Comment (Shwetabhk) [reply apps/fyle/serializers.py:53]: It is not a validation error at all right, it's an unexpected error

File: apps/fyle/serializers.py (hunk 5 above)
Comment (Shwetabhk) [reply apps/fyle/serializers.py:14]: Yes always 2 line spaces between all statements in the top level
there can be only 1 line change within Class functions
```python
import xyz


x = y


class ABC:
    def a():
        pass

    def b():
        pass

    def c():
        pass
```

File: apps/fyle/serializers.py (hunk 6 above)
Comment (Shwetabhk) [reply apps/fyle/serializers.py:53]: Please fix it off

**Review by Shwetabhk — CHANGES_REQUESTED** [review]

File: apps/fyle/urls.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+1. Import the include() function: from django.urls import include, path
+2. Add a URL to urlpatterns: path('blog/', include('blog.urls'))
+"""
+
+from django.urls import path
+from apps.fyle.views import ImportFyleAttributesView
+
+
+urlpatterns = [
+    path('<int:workspace_id>/import_attributes/', ImportFyleAttributesView.as_view(), name='import-fyle-attributes'),
```
Comment (NileshPant1999) [inline apps/fyle/urls.py:22]: @ruuushhh workspace_id is not needed here no ??

**Review by Shwetabhk — APPROVED** [review]

File: apps/workspaces/serializers.py [hunk 8]
```diff
@@ -223,3 +224,28 @@ def create(self, validated_data):
...
+    Workspace Admin Serializer
+    """
+    admin_emails = serializers.SerializerMethodField()
+
+    def get_admin_emails(self, validated_data):
+        """
+        Get Workspace Admins
+        """
+        workspace_id = self.context['request'].parser_context.get('kwargs').get('workspace_id')
+        workspace = Workspace.objects.get(pk=workspace_id)
```
Comment (NileshPant1999) [inline apps/workspaces/serializers.py:240]: why are we doing this @ruuushhh

Comment (ruuushhh) [reply apps/workspaces/serializers.py]: image
to get workspace users @NileshPant1999

File: apps/workspaces/models.py [hunk 9]
```diff
@@ -167,7 +179,7 @@ class ExportSetting(BaseModel):
     )
     default_vendor_name = StringNullField(help_text='default Vendor Name')
     default_vendor_id = StringNullField(help_text='default Vendor Id')
-    auto_create_vendor = BooleanFalseField(help_text='Auto create vendor')
```
Comment (NileshPant1999) [inline apps/workspaces/models.py:170]: @ruuushhh why is this removed.

**Review by Shwetabhk — APPROVED** [review]

**Pull request merged**